### DISCUSSION TITLE
Cache for resources

### DISCUSSION THREAD

**Discussion by DaspawnW** [fabric8io/kubernetes-client#3638] [Q&A] [answered]
Dear fabric8 community,

First of all thanks a lot for the great client :-) I'm using it now since multiple years and I'm very happy with it.

tl;dr;

> Is there any way of a cache which holds a "copy" of the secrets, which maybe simply utilizes the sharedinformerfactory to update the cache if there is any change?

I currently run into a problem, where I would kindly ask for your support:

Based on a custom resource I automatically sync secrets stored in HC Vault with Kubernetes.
Therefore I take the secret out of HC Vault and store it as a secret in Kubernetes. Now in a specified interval I recheck if the secret stored in HC Vault is still in sync. So I take the secret from HC Vault and compare it with the secret in Kubernetes, if it's newer I update it in Kubernetes.

After some time I have now a high number of requests to my Kubernetes API. So the first change I did I switched to an sharedInformerFactory for my custom resource, which is cool as I now can use for this the resync Interval, so no list all custom resource entries and compare the value out of HC Vault with the value out of the secret :-)

But...

I still have to make a high number of requests to get the secrets stored in Kubernetes and this is quiet expensive to the api server [yes you maybe ask: is it really high? I'm talking during this run of around 3000 secrets; and the logic in kubernetes apiserver plays against me: take it from etcd; decrypt the secret; send to client].
Is there any way of a cache which holds a "copy" of the secrets, which maybe simply utilizes the sharedinformerfactory to update the cache if there is any change?

**Comment by manusa** [comment, marked as answer]
I'm not sure I understand correctly, but Why aren't you using a shared informer for the secrets too?
You will have a cached version of the Secret, if the Secret is updated/removed then you will also get notified.

**Reply by DaspawnW** [reply]
Exactly, my question was if there is maybe already a solution for it ;-)
